service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -125,84 +120,99 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 
 				blocks := azureCluster.Spec.NetworkSpec.Vnet.CIDRBlocks
 				if len(blocks) > 0 {
-					clusterValues["clusterCIDR"] = blocks[0]
+					clusterCIDR = blocks[0]
 				}
-			case "GCPCluster":
-				clusterValues["provider"] = "gcp"
-
+			case "aws":
+			case "openstack":
+			case "vsphere":
+				fallthrough
+			case "gcp":
```

Comment:
This isn't correct is it? Nothing for aws and openstack but vsphere falls through to gcp? :/

Reply:
You are right. Fixed, please take a look.